### ISSUE TITLE
The reserve extends beyond the boundaries of the outline

### ISSUE THREAD

**Issue by snooppr** [osmandapp/OsmAnd#19469]
### Description

https://www.openstreetmap.org/relation/8012595#map=14/55.3886/41.9491

[Here is the border of the reserve](http://www.oopt.aari.ru/oopt/%D0%9E%D0%BA%D1%81%D0%BA%D0%B8%D0%B9-%D0%B1%D0%B5%D1%80%D0%B5%D0%B3%D0%BE%D0%B2%D0%BE%D0%B9/map): "Oksky Coast": 

On the osm.org maps, the border seems to be fine (but when editing the view, it seems that the border of the reserve has been broken?)

![Osmand](https://github.com/osmandapp/OsmAnd/assets/61022210/f84e4b95-e589-41ab-8ca5-94809e02df62)

### Steps to reproduce

Update the maps of the Vladimir region or the Central Federal District region before April, the problem will come out.

### Actual result

The reserve extends beyond its borders along its entire length from the village of Repino to the village of Bolshoi Sanchur.

### Expected result

Return the display of the reserve to its borders.

@xmd5a2 Perhaps he will check the maps and correct the boundaries if there is a problem with the boundaries.

### Your Environment (required)

**WARNING** Crash-Logs **MAY** contain information you deem sensitive.
Review this **CAREFULLY** before posting your issue!

```
OsmAnd Version: f-droid
Android/iOS version:
Device model:
Crash-Logs: ?
```

**Comment by yuriiurshuliak** [comment]
The bug was not reproduced. The boundaries of the reserve in the Osmand App align with those outlined in the OSRM. Consider updating your maps.

Here is the link to OSRM: [OSRM](https://www.openstreetmap.org/relation/8012595#map=12/55.3398/42.0176)
And here is the link to the web version: [Web Version](https://test.osmand.net/map/?start=55.334155,41.928163&profile=car#14/55.3331/41.9275)

OsmAnd~ 4.7.0#1982m, released: 2024-04-04

<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd/assets/127092082/607f16a2-e873-4aa5-81b8-505c2898aeae"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd/assets/127092082/b1955ab5-1639-4e97-8275-f8208971e27a"  width="300" height="auto"/> </th>
  </tr>
</table>

**Comment by snooppr** [comment]
@yuriiurshuliak I found what the problem is. Now check the bug reproduction.

The problem is in the maps, but not in the “Vladimir region”, but in the neighboring region: “Nizhny Novgorod region”.

These areas will be separated by the Oka River (the border runs strictly along the river). But the “Nizhny Novgorod region” goes beyond its borders by several kilometers and captures, superimposing on the “Vladimir region”.

To reproduce the bug:
1) Download two regions: “Vladimirskaya” and “Nizhny Novgorod”. You will see the problem that the reserve goes beyond its boundaries.
2) If you disable (archive the Vladimir region), you will see that the Nizhny Novgorod region captures part of the Vladimir region (For example, the village "Lyahi" is displayed, although it is the "Vladimir region" and not the "Nizhny Novgorod region").
3) If you disable (archive the Nizhny Novgorod region) and activate the Vladimir region, then the reserve will be normal within its borders and there will be no bug.

Total: maps of the neighboring regions of Vladimir and Nizhny Novgorod regions conflict with each other (the problem appeared in April, these are the latest maps).

**Comment by snooppr** [comment]
**Additional Information:**

The bug does not appear on all map styles.  There is no bug on map styles ("offroad" and "topo").

![osmand](https://github.com/osmandapp/OsmAnd/assets/61022210/4f31e2e0-3d56-4148-a77c-5a0f2a5cc349)

**Comment by snooppr** [comment]
This was fixed in the osmand update to v 4.7.10.
